code/cabal-audit/src/Distribution/Audit.hs

```diff
@@ -0,0 +1,246 @@
+      `catch` \ex -> throwIO $ CabalException {reason = "elaborating the install-plan", cabalException = ex}
+
+  when (verbosity > Verbosity.normal) do
+    putStrLn (formatWith [blue] "Finished building the cabal install plan, looking for advisories...")
+
+  advisories <- do
+    let k realPath =
+          listAdvisories realPath
+            >>= validation (throwIO . ListAdvisoryValidationError realPath) pure
+    case advisoriesPathOrURL of
+      Left fp -> k fp
+      Right url -> withSystemTempDirectory "cabal-audit" \tmp -> do
+        putStrLn $ formatWith [blue] $ "trying to clone " <> url
+        callProcess "git" ["clone", url, tmp]
+        k tmp
+
+  pure $ matchAdvisoriesForPlan plan advisories
+
+-- | provides the built advisories in some consumable form, e.g. as human readable form
+--
+-- FUTUREWORK(mangoiv): provide output as JSON
+handleBuiltAdvisories :: Output -> M.Map PackageName ElaboratedPackageInfoAdvised -> IO ()
+handleBuiltAdvisories = \case
+  HumanReadable -> humanReadableHandler . M.toList
+  Osv fp -> osvHandler fp
+
+osvHandler :: FilePath -> M.Map PackageName ElaboratedPackageInfoAdvised -> IO ()
+osvHandler fp =
+  Aeson.encodeFile @Value fp . object . M.foldMapWithKey
+    \pn MkElaboratedPackageInfoWith {elaboratedPackageVersion, packageAdvisories} ->
+      [ fromString (unPackageName pn)
+          .= object
+            [ "version" .= prettyVersion @Text elaboratedPackageVersion
+            , "advisories" .= map (OSV.convert . fst) (runIdentity packageAdvisories)
+            ]
+      ]
+
+-- | pretty-prints a `Version`
+--
+-- >>> import Distribution.Version
+-- >>> prettyVersion $ mkVersion [0, 1, 0, 0]
+-- "0.1.0.0"
+prettyVersion :: IsString s => Version -> s
+prettyVersion = fromString . List.intercalate "." . map show . versionNumbers
+{-# INLINE prettyVersion #-}
+
+prettyAdvisory :: Advisory -> Maybe Version -> Text
+prettyAdvisory Advisory {advisoryId, advisoryPublished, advisoryKeywords, advisorySummary} mfv =
+  T.unlines do
+    let hsecId = T.pack (printHsecId advisoryId)
+    map
+      ("  " <>)
+      [ formatWith [bold, blue] hsecId <> " \"" <> advisorySummary <> "\""
+      , "published: " <> formatWith [bold] (T.pack $ show advisoryPublished)
+      , "https://haskell.github.io/security-advisories/advisory/" <> hsecId
+      , fixAvailable
+      , formatWith [blue] $ T.intercalate ", " (coerce advisoryKeywords)
+      ]
+ where
+  fixAvailable = case mfv of
+    Nothing -> formatWith [bold, red] "No fix version available"
+    Just fv -> formatWith [bold, green] "Fix available since version " <> formatWith [yellow] (prettyVersion fv)
+
+-- | this is handler is used when displaying to the user
+humanReadableHandler :: [(PackageName, ElaboratedPackageInfoAdvised)] -> IO ()
+humanReadableHandler = \case
+  [] -> putStrLn (formatWith [green, bold] "No advisories found.")
+  avs -> do
+    putStrLn (formatWith [bold, red] "\n\nFound advisories:\n")
```

Comment:
Does `formatWith` work correctly on windows (e.g. powershell)?

Reply:
I don't have a windows machine to test but these are simple ansi escape codes afaik, I would be surprised if they wouldn't work on windows. I want to add a `--no-color` flag but that will only happen after refactoring to not having everything in `IO` and then just specifying a handler that will decide on whether you do coloured or uncoloured output.

Reply:
Yes. Also see https://no-color.org/

So you should honor the `NO_COLOR` env var.

Reply:
thank you, I will add this as a bullet point to the tracking issue.